docker/algod/source/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM golang:1.13.12
```

Comment:
You could actually utilize the other algod docker image in this one to make things a little faster using multi-stage builds like this:
```
FROM golang:1.13.12 as builder

RUN #Build binaries

FROM algod:latest
COPY --from=builder /algod/bin /algod/bin
```

Doing it this would reduce the amount of duplicated code to keep up with and make recurring builds a bit faster. It would requiring ensuring the algod image is available prior to building this one though.

Reply:
Cool, I didn't know you could combine multiple dockerfiles.

Just so I'm clear, this would require 3 dockerfiles? There would be separate dockerfiles to install from source, or from a channel, and then a 3rd to setup / configure the network?

Reply:
So this isn't actually combining dockerfiles. What this is doing is executing two docker build stages. First it uses the golang base image to build algorand binaries, then in the second stage it copies the binaries that were built onto the already built algod image. Multi-stage docker builds produce images for each stage, but only the last image is tagged. So with this example we would have one docker image which is the algod image with it's binaries overwritten. So it's basically taking the docker image you've already made and overwriting the binaries with ones built from source. It would be two dockerfiles though. One for this process, and another for the algod image you've already written. Does that make sense?

Here's a doc that might help explain it
https://docs.docker.com/develop/develop-images/multistage-build/

Reply:
Thanks for the explanation, that does make sense. I actually did a crude version of this in the past by copying a locally build `algod` binary into the container.

I don't see how this is faster though, either way the private network needs to be created using the new binaries.